File: yilu_an_frontend/miniprogram/api/destination.ts
// 目的地相关接口
import { api } from '../utils/request';

// 创建目的地请求参数
export interface CreateDestinationParams {
  name: string;
  address: string;
  latitude?: number;
  longitude?: number;
  is_common?: boolean;
}

// 目的地
export interface Destination {
  id: number;
  user_id: number;
  name: string;
  address: string;
  latitude: number;
  longitude: number;
  is_common: boolean;
  created_at: string;
}

// 目的地相关API
export const destinationApi = {
  // 获取目的地列表
  getDestinations: () => {
    return api.get<Destination[]>('/api/v1/destinations/');
  },
  
  // 创建目的地
  createDestination: (params: CreateDestinationParams) => {
    return api.post<Destination>('/api/v1/destinations/', params);
  },
};

File: yilu_an_backend/app/models/destination.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Boolean, Float
from sqlalchemy.orm import relationship
from app.database import Base
from datetime import datetime

class Destination(Base):
    __tablename__ = "destinations"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    name = Column(String(100), nullable=False)
    address = Column(String(255), nullable=False)
    latitude = Column(Float)
    longitude = Column(Float)
    is_common = Column(Boolean, default=False)
    created_at = Column(DateTime, default=datetime.now)
    
    user = relationship("User", back_populates="destinations")
